scripts/ASTImportTest.sh

```diff
@@ -1,105 +1,190 @@
-            # first failing test
-            # exit 1
-            FAILED=$((FAILED + 1))
-            echo -e "ERROR: AST reimport failed for input file $nth_input_file"
-            echo
-            echo "Compiler stderr:"
-            cat ./stderr.txt
-            echo
-            echo "Compiler stdout:"
-            cat ./obtained.json
+            print_used_commands "$(pwd)" "${export_command[*]}" "${import_command[*]}"
             return 1
         fi
-        DIFF="$(diff expected.json obtained.json)"
-        if [ "$DIFF" != "" ]
-        then
-            if [ "$DIFFVIEW" == "" ]
-            then
-                echo -e "ERROR: JSONS differ for $1: \n $DIFF \n"
-                echo "Expected:"
-                cat ./expected.json
-                echo "Obtained:"
-                cat ./obtained.json
-            else
-                # Use user supplied diff view binary
-                $DIFFVIEW expected.json obtained.json
-            fi
-            FAILED=$((FAILED + 1))
-            return 2
-        fi
-        TESTED=$((TESTED + 1))
-        rm expected.json obtained.json
-        rm -f stderr.txt
+        FAILED=$((FAILED + 1))
+    fi
+    TESTED=$((TESTED + 1))
+}
+
+# function tests whether exporting and importing again is equivalent.
+# Results are recorded by adding to FAILED or UNCOMPILABLE.
+# Also, in case of a mismatch a diff is printed
+# Expected parameters:
+# $1 name of the file to be exported and imported
+# $2 any files needed to do so that might be in parent directories
+function test_import_export_equivalence {
+    local sol_file="$1"
+    local input_files=( "${@:2}" )
+    local OUTPUT
+    local SOLC_RC
+
+    set +e
+    OUTPUT=$("$SOLC" --bin "${input_files[@]}" 2>&1)
+    SOLC_RC=$?
+    set -e
+
+    # if input files where compilable with success
+    if (( SOLC_RC == 0 ))
+    then
+        case "$IMPORT_TEST_TYPE" in
+            ast) ast_import_export_equivalence "${sol_file}" "${input_files[@]}" ;;
+            *) fail "Unknown import test type '${IMPORT_TEST_TYPE}'. Aborting." ;;
+        esac
     else
-        # echo "contract $solfile could not be compiled "
         UNCOMPILABLE=$((UNCOMPILABLE + 1))
+        # solc will return exit code 2, if it was terminated by an uncaught exception.
+        if (( SOLC_RC == 2 ))
+        then
+          # Ignore all UnimplementedFeatureError exceptions.
+          if [[ "$OUTPUT" != *"UnimplementedFeatureError"* ]]
+          then
+            printError "\n\nERROR: Uncaught Exception while executing '$SOLC --bin ${input_files[*]}':\n${OUTPUT}\n"
+            exit 1
+          fi
+        fi
     fi
-    # return 0
 }
-echo "Looking at $NSOURCES .sol files..."
 
 WORKINGDIR=$PWD
 
-# for solfile in $(find $DEV_DIR -name *.sol)
+command_available "$SOLC" --version
+command_available jq --version
+
+case "$IMPORT_TEST_TYPE" in
+    ast) TEST_DIRS=("${SYNTAXTESTS_DIR}" "${ASTJSONTESTS_DIR}") ;;
+    *)  print_usage ;;
```

Comment:
Please use `assertFail` here. This was already validated so there should be no unexpected values here.